test/lint/lint-includes.py

```diff
@@ -21,17 +21,23 @@
                  "src/minisketch/",
                 ]
 
-EXPECTED_BOOST_INCLUDES = ["boost/date_time/posix_time/posix_time.hpp",
+EXPECTED_BOOST_INCLUDES = ["boost/assert.hpp",
+                           "boost/date_time/posix_time/posix_time.hpp",
                            "boost/multi_index/hashed_index.hpp",
+                           "boost/multi_index/identity.hpp",
+                           "boost/multi_index/indexed_by.hpp",
                            "boost/multi_index/ordered_index.hpp",
                            "boost/multi_index/sequenced_index.hpp",
+                           "boost/multi_index/tag.hpp",
```

Comment:
I would expect to see `boost/multi_index_container.hpp` removed from this list as part of this PR, but it looks like it's still used:
```bash
 $ git grep multi_index_container.hpp origin/pr/27783
origin/pr/27783:node/miner.h:#include <boost/multi_index_container.hpp>
origin/pr/27783:txmempool.h:#include <boost/multi_index_container.hpp>
origin/pr/27783:txrequest.cpp:#include <boost/multi_index_container.hpp>
```

Is there some complication with those 3 that prevents us from using the specific headers rather than the umbrella one?

Reply:
Nevermind, I misremembered how `multi_index_container.hpp` worked. I was thinking it was just a dumb list of other includes, but it's more than that.

Still, does using `multi_index_container_fwd.hpp` help at all?

Reply:
> Still, does using `multi_index_container_fwd.hpp` help at all?

No, it doesn't:
```
./txmempool.h:406:29: error: field ‘mapTx’ has incomplete type ‘CTxMemPool::indexed_transaction_set’ {aka ‘boost::multi_index::multi_index_container<CTxMemPoolEntry, boost::multi_index::indexed_by<boost::multi_index::hashed_unique<mempoolentry_txid, SaltedTxidHasher>, boost::multi_index::hashed_unique<boost::multi_index::tag<index_by_wtxid>, mempoolentry_wtxid, SaltedTxidHasher>, boost::multi_index::ordered_non_unique<boost::multi_index::tag<descendant_score>, boost::multi_index::identity<CTxMemPoolEntry>, CompareTxMemPoolEntryByDescendantScore>, boost::multi_index::ordered_non_unique<boost::multi_index::tag<entry_time>, boost::multi_index::identity<CTxMemPoolEntry>, CompareTxMemPoolEntryByEntryTime>, boost::multi_index::ordered_non_unique<boost::multi_index::tag<ancestor_score>, boost::multi_index::identity<CTxMemPoolEntry>, CompareTxMemPoolEntryByAncestorFee> > >’}
  406 |     indexed_transaction_set mapTx GUARDED_BY(cs);
      |   
```